src/cache/utils.py

```
@@ -0,0 +1,72 @@
+            return self.get_workspace_cache_key()
+
+    @classmethod
+    def from_cache_key(cls, cache_key: str) -> "CacheKey":
+        """Parse a cache key and extract the workspace name and resource type."""
+        parts = cache_key.split(":")
+        if (
+            len(parts) < 3
+            or parts[0] != settings.CACHE.NAMESPACE
+            or parts[1] != "workspace"
+        ):
+            raise ValueError(f"Invalid cache key format: {cache_key}")
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)